File: 1376-time-needed-to-inform-all-employees/1376-time-needed-to-inform-all-employees.cpp
class Solution {
public:
    int ans = 0;
    vector<int> graph[100003];
    void dfs(int parent , int time , vector<int>& informTime){
        for(auto child : graph[parent]){
            dfs(child , time+informTime[child] , informTime);
        }
        ans = max(ans , time);
    }
    int numOfMinutes(int n, int headID, vector<int>& manager, vector<int>& informTime) {
        for(int i=0;i<n;++i){
            if(i != headID){
                graph[manager[i]].push_back(i);
            }
        }
        dfs(headID , informTime[headID] ,informTime);
        return ans;
    }
};

File: 1376-time-needed-to-inform-all-employees/1376-time-needed-to-inform-all-employees.java
class Solution {
    private List<List<Integer>> graph = new ArrayList<>();
    private int ans = 0;
    void dfs(int parent , int time , int[] informTime){
        for(int i=0;i<graph.get(parent).size();++i){
            dfs(graph.get(parent).get(i) , time+informTime[graph.get(parent).get(i)] , informTime);
        }
        ans = Math.max(ans , time);
    }
    public int numOfMinutes(int n, int headID, int[] manager, int[] informTime) {
        for(int i=0;i<n;++i){
            graph.add(new ArrayList<>());
        }
        for(int i=0;i<n;++i){
            if(i != headID){
                graph.get(manager[i]).add(i);
            }
        }
        dfs(headID , informTime[headID] , informTime);
        return ans;
    }
}